latency-breakdown.md

```diff
@@ -0,0 +1,869 @@
+
+- The following is the time cost diagram of async write operations when the asynchronous IO is disabled:
+
+    ```railroad+diagram
+    Diagram(
+        NonTerminal("Propose Wait"),
+        NonTerminal("Process Command"),
+        Choice(
+            0,
+            Sequence(
+                NonTerminal("Wait Current Batch"),
+                NonTerminal("Write to Log Engine"),
+            ),
+            Sequence(
+                NonTerminal("RaftMsg Send Wait"),
+                NonTerminal("Commit Log Wait"),
+            ),
+        ),
+        NonTerminal("Apply Wait"),
+        NonTerminal("Apply Log"),
+    )
+    ```
+
+- The following is the time cost diagram of async write operations when the asynchronous IO is enabled:
+
+    ```railroad+diagram
+    Diagram(
+        NonTerminal("Propose Wait"),
+        NonTerminal("Process Command"),
+        Choice(
+            0,
+            NonTerminal("Wait Until Persisted by Write Worker"),
+            Sequence(
+                NonTerminal("RaftMsg Send Wait"),
+                NonTerminal("Commit Log Wait"),
+            ),
+        ),
+        NonTerminal("Apply Wait"),
+        NonTerminal("Apply Log"),
+    )
+    ```
+
+The async write duration is calculated as:
+
+```text
+async write duration(async io disabled) =
+    propose +
+    async io disabled commit +
+    tikv_raftstore_apply_wait_time_duration_secs +
+    tikv_raftstore_apply_log_duration_seconds
+
+async write duration(async io enabled) =
+    propose +
+    async io enabled commit +
+    tikv_raftstore_apply_wait_time_duration_secs +
+    tikv_raftstore_apply_log_duration_seconds
+```
+
+Async write can be broken down into the following three phases:
+
+- Propose
+- Commit
+- Apply: `tikv_raftstore_apply_wait_time_duration_secs + tikv_raftstore_apply_log_duration_seconds` in the preceding formula
+
+The duration of propose phase is calculated as:
+
+```text
+propose =
+    propose wait duration +
+    propose duration
+
+propose wait duration =
+    tikv_raftstore_store_wf_batch_wait_duration_seconds
+
+propose duration =
+    tikv_raftstore_store_wf_send_to_queue_duration_seconds -
+    tikv_raftstore_store_wf_batch_wait_duration_seconds
+```
+
+The Raft process is recorded in a waterfall manner. So the propose duration is calculated from the difference between the two metrics.
+
+The duration of commit phase is calculated as:
+
+```text
+async io disabled commit = max(
+    persist log locally duration,
+    replicate log duration
+)
+
+async io enabled commit = max(
+    wait by write worker duration,
+    replicate log duration
+)
+```
+
+Since v5.3.0, TiKV supports Async IO Raft(write Raft log by a StoreWriter thread pool), which is only enabled when setting the [`store-io-pool-size`](/tikv-configuration-file.md#store-io-pool-size-new-in-v530) as a positive value, which changes the process of commit. The `persist log locally duration` and `wait by write worker duration` are calculated as:
```

Comment:
Please do not chain multiple "which"es, which leads to a long-winded sentence.
```suggestion
Since v5.3.0, TiKV supports Async IO Raft (write Raft log by a StoreWriter thread pool). The Async IO Raft is only enabled when the [`store-io-pool-size`](/tikv-configuration-file.md#store-io-pool-size-new-in-v530) is set to a positive value, which changes the process of commit. The `persist log locally duration` and `wait by write worker duration` are calculated as:
```